sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -383,14 +383,20 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
         // Get all input data source relations of the query.
         val srcRelations = df.logicalPlan.collect {
           case LogicalRelation(src: BaseRelation, _, _) => src
+          case relation: CatalogRelation if DDLUtils.isHiveTable(relation.catalogTable) =>
+            relation.catalogTable.identifier
         }
         EliminateSubqueryAliases(catalog.lookupRelation(tableIdentWithDB)) match {
           // Only do the check if the table is a data source table (the relation is a BaseRelation).
```

Comment:
this comment also need to be updated, we do not "only" check data source tables.